crawl-ref/source/spl-transloc.cc

```diff
@@ -1044,6 +1047,90 @@ spret cast_portal_projectile(int pow, bool fail)
     return spret::success;
 }
 
+static bool _projectable_weapon()
+{
+    if (!you.weapon())
+        return true;
+    const item_def &it = *you.weapon();
+    // These all cause attack prompts, which are awkward to handle.
+    // TODO: support these!
+    static const vector<int> forbidden_unrands = {
+        UNRAND_POWER,
+        UNRAND_DEVASTATOR,
+        UNRAND_VARIABILITY,
+        UNRAND_SINGING_SWORD,
+        UNRAND_TORMENT,
+        UNRAND_ARC_BLADE,
+    };
+    for (int urand : forbidden_unrands)
+        if (is_unrandom_artefact(it, urand))
+            return false;
+    return true;
+}
+
+spret cast_manifold_assault(int pow, bool fail, bool real)
+{
+    vector<monster*> targets;
+    for (monster_near_iterator mi(&you, LOS_NO_TRANS); mi; ++mi)
+    {
+        if (mi->wont_attack() || mi->neutral())
+            continue; // this should be enough to avoid penance?
+        if (!you.can_see(**mi))
+            continue;
+        targets.emplace_back(*mi);
+    }
+
+    if (targets.empty())
+    {
+        if (real)
+            mpr("You can't see anything to attack.");
+        return spret::abort;
+    }
+
+    if (!_projectable_weapon())
+    {
+        if (real)
+        {
+            mprf("%s would react catastrophically with paradoxical space!",
+                 you.weapon()->name(DESC_THE, false, false, false, false, ISFLAG_KNOW_PLUSES).c_str());
+        }
+        return spret::abort;
+    }
+
+    if (!real)
+        return spret::success;
+
+    if (!wielded_weapon_check(you.weapon()))
+        return spret::abort;
+
+    fail_check();
+
+    mpr("Space warps momentarily into an impossible shape!");
+
+    const int initial_time = you.time_taken;
+
+    shuffle_array(targets);
+    const size_t max_targets = 2 + div_rand_round(pow, 50);
+    for (size_t i = 0; i < max_targets && i < targets.size(); i++)
+    {
+        // manually apply noise
+        behaviour_event(targets[i], ME_ALERT, &you, you.pos()); // shout + set you as foe
+
+        // Somewhat hacky: reset attack delay before each attack so that only the final
+        // attack ends up actually setting time taken. (No quadratic effects.)
```

Comment:
Maybe include a mention of the XXX in attack_delay, so if someone fixes that there's hope that this hack gets removed too

Reply:
Good idea. Done.